### PR TITLE
Update support policy docs

### PR DESCRIPTION
**Pull request by mthalman**
* Clarified the documentation to explicitly state that unlisted tags are not supported.
* Renamed tagging-guidelines.md to supported-tags.md. Git doesn't handle it as a rename (too different). The differences start at line 55.
* Added link to the Supported Tags doc in the support section of the readmes.
* Added Tag Listing, Tag Lifecycle, and Policy Changes sections to the Supported Tags doc. Puts it in line with https://github.com/dotnet/dotnet-docker.

### CODE REVIEW

File: documentation/supported-tags.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Tags | Dockerfile | OS Version
+-----------| -------------| -------------
+4.8-20220412-windowsservercore-ltsc2022, 4.8-windowsservercore-ltsc2022, 4.8, latest | [Dockerfile](https://github.com/microsoft/dotnet-framework-docker/blob/228ad35840777951381456d3f89392c3a8fe8a89/src/sdk/4.8/windowsservercore-ltsc2022/Dockerfile)
+
+## Tag Lifecycle
+
+Except for tags containing a timestamp, each tag will be supported for the lifetime of the .NET Framework and Windows Server version referenced by the tag. Once either either of these reaches EOL, the tag will be considered unsupported, will no longer be updated and will be removed from the [Tag Listing](#tag-listing). Tags containing a timestamp are only supported until a newer timestamp is published for that same .NET Framework and Windows Server version.
```
Comment (MichaelSimons) [inline documentation/supported-tags.md:65]:
```suggestion
Except for tags containing a timestamp, each tag will be supported for the lifetime of the .NET Framework and Windows Server version referenced by the tag. Once either of these reaches EOL, the tag will be considered unsupported, will no longer be updated and will be removed from the [Tag Listing](#tag-listing). Tags containing a timestamp are only supported until a newer timestamp is published for that same .NET Framework and Windows Server version.
```

File: documentation/supported-tags.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Unsupported tags will be preserved to prevent breaking any references to it.
+
+### Examples
+
+* `4.8` - Will be supported for the lifetime of the .NET Framework 4.8 release.
+* `4.8-windowservercore-ltsc2022` - Will be supported for the lifetime of .NET Framework 4.8 and Windows Server Core LTSC 2022 releases, whichever is shorter.
+* `4.8-20220412-windowsservercore-ltsc2022` - Will be supported only until a newer timestamp is published for that same .NET Framework and Windows Server version.
```
Comment (MichaelSimons) [inline documentation/supported-tags.md:73]: What do you think about calling out the typical lifespan is a month?

File: eng/readme-templates/README.md [hunk 3]
```diff
@@ -58,6 +58,9 @@ The [.NET Framework Docker samples](https://github.com/microsoft/dotnet-framewor
 * [dotnet/samples](https://hub.docker.com/_/microsoft-dotnet-samples/): .NET Samples
 
 {{if !IS_PRODUCT_FAMILY:# Full Tag Listing
+<!--End of generated tags-->
+
+*Tags not listed in the table above should be considered unsupported.*
```
Comment (MichaelSimons) [inline eng/readme-templates/README.md:63]: What's the reason for capitalizing 'Listed'?
Consider using stronger language than "should be" - they are definitely unsupported.

Tags excluded from the above table are unsupported.  See the [Supported Tags Policy](https://github.com/microsoft/dotnet-framework-docker/blob/main/documentation/supported-tags.md).

File: documentation/supported-tags.md (hunk 1 above)
Comment (MichaelSimons) [inline documentation/supported-tags.md:65]: > Except for tags containing a timestamp

Another exception is `latest`

File: eng/readme-templates/README.md [hunk 4]
```diff
@@ -78,6 +81,8 @@ Version Tag | OS Version | Supported .NET Versions
 
 See the [.NET Framework Lifecycle FAQ](https://support.microsoft.com/help/17455/lifecycle-faq-net-framework)
 
+See detailed support information for .NET Frameworks containers on [tags](https://github.com/microsoft/dotnet-framework-docker/blob/main/documentation/supported-tags.md).
```
Comment (MichaelSimons) [inline eng/readme-templates/README.md:84]: What do you think about making this a list like the Image Update Policy, Feedback and License sections?

* [.NET Framework Lifecycle FAQ](https://support.microsoft.com/help/17455/lifecycle-faq-net-framework)
* [Supported Tags Policy](https://github.com/microsoft/dotnet-framework-docker/blob/main/documentation/supported-tags.md)

File: eng/readme-templates/README.md (hunk 3 above)
Comment (mthalman) [reply eng/readme-templates/README.md:63]: > What's the reason for capitalizing 'Listed'?

It's not. That's just the italic font.

File: eng/readme-templates/README.md [hunk 5]
```diff
@@ -58,6 +58,9 @@ The [.NET Framework Docker samples](https://github.com/microsoft/dotnet-framewor
 * [dotnet/samples](https://hub.docker.com/_/microsoft-dotnet-samples/): .NET Samples
 
 {{if !IS_PRODUCT_FAMILY:# Full Tag Listing
+<!--End of generated tags-->
+
+*Tags not listed in the table above are not unsupported. See the [Supported Tags Policy](https://github.com/microsoft/dotnet-framework-docker/blob/main/documentation/supported-tags.md).*
```
Comment (MichaelSimons) [inline eng/readme-templates/README.md:63]: This wording feels like a double negative - not unsupported.

File: documentation/supported-tags.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+Tags | Dockerfile | OS Version
+-----------| -------------| -------------
+4.8-20220412-windowsservercore-ltsc2022, 4.8-windowsservercore-ltsc2022, 4.8, latest | [Dockerfile](https://github.com/microsoft/dotnet-framework-docker/blob/228ad35840777951381456d3f89392c3a8fe8a89/src/sdk/4.8/windowsservercore-ltsc2022/Dockerfile)
+
+## Tag Lifecycle
+
+Each tag will be supported for the lifetime of the .NET Framework and Windows Server version referenced by the tag. Once either of these reaches EOL, the tag will be considered unsupported, will no longer be updated and will be removed from the [Tag Listing](#tag-listing). There are two exceptions:
+
+* Tags containing a timestamp are only supported until a newer timestamp is published for that same .NET Framework and Windows Server version. The timeframe for this is typically on a monthly basis on ["Patch Tuesday"](https://www.microsoft.com/msrc/faqs-security-update-guide).
+* Support of the `latest` tag will match the support lifetime of the latest version of .NET Framework.
```
Comment (MichaelSimons) [inline documentation/supported-tags.md:68]: This wording is not clear to me.  What about something like

> The `latest` tag will always reference the latest version of .NET Framework.

File: README.wcf.md [hunk 7]
```diff
@@ -78,6 +78,9 @@ Tag | Dockerfile
 4.6.2-20220412-windowsservercore-ltsc2016, 4.6.2-windowsservercore-ltsc2016, 4.6.2 | [Dockerfile](https://github.com/microsoft/dotnet-framework-docker/blob/main/src/wcf/4.6.2/windowsservercore-ltsc2016/Dockerfile)
 
 You can retrieve a list of all available tags for dotnet/framework/wcf at https://mcr.microsoft.com/v2/dotnet/framework/wcf/tags/list.
+<!--End of generated tags-->
```
[the remark this answers is not in the document]
Comment (mthalman) [reply README.wcf.md:81]: This is necessary for the MCR rendering tool to recognize the end of the section in which to generate the tags table. This is needed if there's additional content that needs to be preserved prior to the next markdown header. It's the same scenario as this: https://github.com/dotnet/dotnet-docker/blob/8010d10e71058c5fe05bf48995300c78410925b1/eng/readme-templates/README.md?plain=1#L86

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**